detox/src/Detox.js

```diff
@@ -94,7 +94,7 @@ class Detox {
       this._client = null;
     }
 
-    if (this.device) {
+    if (this.device && this.device.id) {
```

Comment:
Ensure that a device allocation has taken place.
Otherwise, the non-initialized `device` instance is prone to null-reference errors.